homeassistant/components/apsystemsapi_local/sensor.py

```diff
@@ -0,0 +1,175 @@
+        state_class=SensorStateClass.TOTAL,
+        value_fn=lambda c: c.te1 + c.te2,
+    ),
+    ApsystemsLocalApiSensorDescription(
+        key="lifetime_production_p1",
+        translation_key="lifetime_production_p1",
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL,
+        value_fn=lambda c: c.te1,
+    ),
+    ApsystemsLocalApiSensorDescription(
+        key="lifetime_production_p2",
+        translation_key="lifetime_production_p2",
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL,
+        value_fn=lambda c: c.te2,
+    ),
+    ApsystemsLocalApiSensorDescription(
+        key="today_production",
+        translation_key="today_production",
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        value_fn=lambda c: c.e1 + c.e2,
+    ),
+    ApsystemsLocalApiSensorDescription(
+        key="today_production_p1",
+        translation_key="today_production_p1",
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        value_fn=lambda c: c.e1,
+    ),
+    ApsystemsLocalApiSensorDescription(
+        key="today_production_p2",
+        translation_key="today_production_p2",
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        value_fn=lambda c: c.e2,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: config_entries.ConfigEntry,
+    add_entities: AddEntitiesCallback,
+    discovery_info: DiscoveryInfoType | None = None,
+) -> None:
+    """Set up the sensor platform."""
+    config = hass.data[DOMAIN][config_entry.entry_id]
+    coordinator = config["COORDINATOR"]
+    device_name = config[CONF_NAME]
+
+    add_entities(
+        ApSystemsSensorWithDescription(coordinator, desc, device_name)
+        for desc in SENSORS
+    )
+
+
+class ApSystemsSensorWithDescription(CoordinatorEntity, SensorEntity):
+    """Base sensor to be used with description."""
+
+    entity_description: ApsystemsLocalApiSensorDescription
+
+    def __init__(
+        self,
+        coordinator: ApSystemsDataCoordinator,
+        entity_description: ApsystemsLocalApiSensorDescription,
+        device_name: str,
+    ) -> None:
+        """Initialize the sensor."""
+        super().__init__(coordinator)
+        self._state: int | float | None = None
+        self.entity_description = entity_description
+        self._device_name = device_name
+        self._attr_unique_id = f"apsystemsapi_{device_name}_{entity_description.key}"
```

Comment:
Can device names be changed by the user? If so, they cannot be used as a unique ID.

Reply:
Yeah, looks like device name is provided by the user.

If we cannot access a serial number or mac address from the device, use the config entry ID instead of device name.